lib/FS.ml

```diff
@@ -215,6 +276,23 @@ module Make (IO: IO) = struct
           SHA.of_hex p
         ) packs in
       Lwt.return packs
+*)
```

Comment:
it's better to read the diffs if you delete the previous code instead of commenting it out. So I'm not sure to understand what the `packs_opt` variable is for. What happens if someone runs `git gc` to create a new pack file behind your back? How do you update your cache of pack hashes?